src/script.js

```diff
@@ -0,0 +1,306 @@
+
+      // If the game has just been started, enable the join integration
+      if (isStarted) {
+        toggleIntegrations(callFrame, Integration.Player);
+        return;
+      }
+
+      // If the game has just been stopped, disable the join
+      // integration and enable the Kahoot host integration.
+      toggleIntegrations(callFrame, Integration.Host);
+    })
+    .on('participant-joined', (e) => {
+      // If this player is a current Kahoot host, disable
+      // the local host integration and enable the player integration.
+      const isHost = !!e.participant.userData?.isKahootHost;
+      if (isHost) {
+        toggleIntegrations(callFrame, Integration.Player);
+      }
+    });
+
+  return callFrame;
+}
+
+/**
+ * Leaves the Daily video call
+ * @param callFrame
+ */
+function leaveCall(callFrame) {
+  const welcomeScreen = document.getElementById('welcomeScreen');
+  const callContainer = document.getElementById('callContainer');
+  callContainer.classList.add(hideClassName);
+  welcomeScreen.classList.remove(hideClassName);
+  callFrame.destroy();
+}
+
+/**
+ * Updates the text of the Kahoot host button to represent current game state
+ * @param callFrame
+ * @param enableStart
+ */
+function updateKahootHostButton(callFrame, enableStart) {
+  const buttons = callFrame.customTrayButtons();
+  if (enableStart) {
+    buttons.kahootHost.label = 'Start Kahoot Game';
+    buttons.kahootHost.tooltip = 'Start Kahoot Game';
+  } else {
+    buttons.kahootHost.label = 'Exit Kahoot';
+    buttons.kahootHost.tooltip = 'Exit Kahoot';
+  }
+  callFrame.updateCustomTrayButtons(buttons);
+}
+
+/**
+ * Updates the local user's Kahoot host state and broadcasts
+ * that state as an app-message to all other participants.
+ * @param callFrame
+ * @param isStartingGame
+ */
+function updateGameHostState(callFrame, isStartingGame) {
+  callFrame.setUserData({
+    isKahootHost: isStartingGame,
+  });
+
+  callFrame.sendAppMessage({ isGameStarted: isStartingGame }, '*');
+}
+
+/**
+ * Gets the current Kahoot host, if any, and returns their Daily session ID
+ * @param callFrame
+ * @returns {undefined|string}
+ */
+function getCurrentKahootHost(callFrame) {
+  const allParticipants = callFrame.participants();
+  const participants = Object.values(allParticipants);
+  for (let i = 0; i < participants.length; i += 1) {
+    const p = participants[i];
+    if (p.userData?.isKahootHost) {
+      return p.session_id;
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Enables the specified integrations (either Kahoot Host or Player),
+ * and disables the other one
+ * @param callFrame
+ * @param integration
+ */
+function toggleIntegrations(callFrame, integration) {
+  console.log('toggling integration', integration);
+  const integrations = callFrame.customIntegrations();
+
+  switch (integration) {
+    case Integration.Host: {
+      // Delete the Player integration and create a Host integration
+      removePlayerIntegration(integrations);
```

Comment:
Pulling out the removal and addition of the host integration to the pre-fetched integration object into more intuitively named functions here